src/components/CurrencyInputPanel/index.js

```diff
@@ -301,6 +316,49 @@ class CurrencyInputPanel extends Component {
     );
   }
 
+  renderTokenButton() {
```

Comment:
This is getting big, let's refactor into it's own own component. It can be put in `src/components/CurrencyInputPanel/TokenButton.js`